File: src/main/java/com/mrkirby153/snowsgivingbot/entity/GiveawayRoleEntity.java
package com.mrkirby153.snowsgivingbot.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "giveaway_roles")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class GiveawayRoleEntity {

    @Id
    @GeneratedValue
    private long id;

    @NonNull
    @Column(name = "guild")
    private String guildId;

    @NonNull
    @Column(name = "role_id")
    private String roleId;
}

File: src/main/kotlin/com/mrkirby153/giveaways/jpa/GiveawayRoleEntity.kt
package com.mrkirby153.giveaways.jpa

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table
import org.springframework.data.jpa.repository.JpaRepository


/**
 * Entity representing a role that has permission to start giveaways
 */
@Table(name = "giveaway_roles")
@Entity
class GiveawayRoleEntity(
    @Column(name = "guild")
    val guildId: String,
    @Column(name = "role_id")
    val roleId: String
) : AutoIncrementingJpaEntity<Long>() {
}

interface GiveawayRoleRepository : JpaRepository<GiveawayRoleEntity, Long> {

}